.github/workflows/sync.yml

```
@@ -0,0 +1,134 @@
+permissions:
+  contents: write
+
+name: API Docs Schema Nightly Sync
+on:
+  schedule:
+    - cron: '0 0 * * *' # every day at midnight UTC
+
+jobs:
+  sync_api_docs_with_binary_websocket_api:
+    name: Nightly Sync API Docs with Binary Websocket API
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+        with:
+          path: deriv-api-docs
+          ref: master
+
+      - name: Checkout binary-websocket-api Repository
+        uses: actions/checkout@v4
+        with:
+          repository: regentmarkets/binary-websocket-api
+          path: deriv-websockets-api
+          token: ${{ secrets.API_SYNC_PAT }} # TODO: Replace with correct Personal Access Token when provided
+          ref: master
+
+      - name: Setup Node
+        uses: actions/setup-node@v4
+        with:
+          node-version: '21'
+
+      - name: Install Dependencies
+        run: |
+          npm install yaml
+          npm install fs-extra
+
+      - name: Configure Git
+        run: |
+          git config --global user.name "PUT NAME HERE WHEN PROVIDED"
+          git config --global user.email "put_email_here@when_provided.com"
+
+      - name: Copy Files From Source to Destination
+        run: node deriv-api-docs/.github/scripts/get-web-socket-api-list.js
+
+      - name: Generate Branch Name
+        id: generateBranchName
+        run: echo "BRANCH=update-schema_$(date +'%Y-%m-%d')" >> $GITHUB_OUTPUT
+
+      - name: Create Pull Request
+        id: createPullRequest
+        uses: peter-evans/create-pull-request@v5
+        with:
+          commit-message: JSON Schema Update
+          title: JSON Schema Update
+          body: JSON Schema Update
+          branch: ${{ steps.generateBranchName.outputs.BRANCH }}
+          base: master
+          token: ${{ secrets.TEST_API_PR_PAT }} # TODO: Replace with correct Personal Access Token when provided
```

```suggestion
          token: ${{ secrets. GITHUB_TOKEN }} # TODO: Replace with correct Personal Access Token when provided
```
I think we can now create PRs from a github action. So, `GITHUB_TOKEN` should be sufficient. Please check with @ali-hosseini-deriv

---

Alright @balakrishna-deriv. The  `GITHUB_TOKEN ` didn't work in my test repo, but I'll use it. If it fails, then we can create one